caikit/runtime/http_server.py

```diff
@@ -272,10 +278,99 @@ def _get_request_params(
+            # get json from pydantic model
+            training_data_json = training_data.model_dump_json()
+            substituted_json = ""
+            # we're asking PYDANTIC_REGISTRY to give us back a pydantic
+            # model of a JsonData, but we first want to get the DM class
+            # for the type of training_data, hence nested calls.
+            if isinstance(
+                training_data.data_stream,
+                PYDANTIC_REGISTRY.get(
+                    PYDANTIC_REGISTRY.get(type(training_data)).JsonData
+                ),
+            ):
+                # substitute data_stream in json repr with jsondata
+                substituted_json = training_data_json.replace("data_stream", "jsondata")
+            elif isinstance(
+                training_data.data_stream,
+                PYDANTIC_REGISTRY.get(PYDANTIC_REGISTRY.get(type(training_data)).File),
+            ):
+                # substitute data_stream in json repr with file
+                substituted_json = training_data_json.replace("data_stream", "file")
+
+            json_data_obj = PYDANTIC_REGISTRY.get(type(training_data)).from_json(
+                substituted_json
+            )
+            request_params["training_data"] = json_data_obj
+
+        # Convert all to DM object types
+        for param_name, param_value in request_params.items():
+            if param_name != "training_data" and issubclass(
+                type(param_value), pydantic.BaseModel
+            ):
+                request_params[param_name] = PYDANTIC_REGISTRY.get(
+                    type(param_value)
+                ).from_json(param_value.model_dump_json())
+        return request_params
+
+    def _train_add_unary_input_unary_output_handler(self, rpc: CaikitRPCBase):
+        """Add a unary:unary request handler for this RPC signature"""
+        pydantic_request = self._dataobject_to_pydantic(
+            self._get_request_dataobject(rpc)
+        )
+        pydantic_response = self._dataobject_to_pydantic(
+            self._get_response_dataobject(rpc)
+        )
+
+        @self.app.post(self._get_route(rpc))
+        # pylint: disable=unused-argument
+        async def _handler(
+            request: pydantic_request, context: Request
+        ) -> pydantic_response:
+            log.debug("In unary handler for %s", rpc.name)
+            loop = asyncio.get_running_loop()
+            request_params = self._get_request_params(rpc, request=request)
+
+            self.build_request_params_dict(request_params)
+            try:
+                call = partial(
+                    self.global_train_servicer.run_training_job,
+                    request=request,
+                    module=rpc.clz,
+                    training_output_dir=None,  # pass None so that GTS picks up the config one # TODO: double-check?
+                    request_params=request_params,
+                    # context=context,
```

Comment:
Need to figure this out. The `context` we expect in `GTS` is different than the `context` sent here.

Reply:
Should be fixed after we do #415 ?

Reply:
Yup, there's some mad refactoring opportunity here.

It's interesting that this http server is actually following the pattern we had wanted for the grpc server but didn't get around to- where individual handlers are built for each `rpc`. This could all be so congruent and beautiful... if we ever have the time to do it 😉